features/atcryptoauth/ATCADevice.h

```diff
@@ -0,0 +1,283 @@
+     *  - Wait Tpu for device Up
+     *  - Pull SDA low for Twlo
+     *  - SDA back high
+     *  - Wait Twhi
+     *  - Ready. A following read should get status 0x11(Wake token received)
+     */
+    ATCAError   Wakeup();
+    /** Reset watchdog timer.
+     *
+     *  Avoid deep sleep and loss of current command state on watchdog timer
+     *  expiry by putting device in Idle mode and straight away waking up.
+     */
+    ATCAError   ResetWatchdog();
+
+    /** Execute a command.
+     *
+     *  This function performs command execution based on datasheet section 9.1:
+     *      - Constructs command tx buffer based on device IO protocol.
+     *      - Transmit tx buffer.
+     *      - Receive response.
+     *      - Check response status.
+     *      - Recover from watchdog expiry or sync failure.
+     *      - Returns response and status back to caller.
+     *
+     *  @param info Command parameters
+     *  @return     Error code from enum ATCAError.
+     */
+    ATCAError   RunCommand(ATCACmdInfo * info);
+
+    /** Execute a read command. This function limits the read data to 4 bytes.
+     *
+     *  Byte address is required to be 4 byte aligned.
+     *
+     *  @param zone     Device EEPROM zone.
+     *  @param address  Byte address within a zone.
+     *  @param word     output buffer of 4 bytes.
+     *  @param wlen     Output buffer length.
+     *  @return         Error code from enum ATCAError.
+     */
+    ATCAError   ReadCommand(ATCAZone zone, uint16_t address,
+                            uint8_t * word, size_t wlen);
+
+    /** Execute a write command. Writes are also limited to 4 bytes.
+     *
+     *  Byte address is required to be 4 byte aligned.
+     *
+     *  @param zone     Device EEPROM zone.
+     *  @param address  Byte address within a zone.
+     *  @param word     input buffer of 4 bytes.
+     *  @param wlen     input buffer length.
+     *  @return         Error code from enum ATCAError.
+     */
+    ATCAError   WriteCommand(ATCAZone zone, uint16_t address,
+                             const uint8_t * word, size_t wlen);
+
+    /** Execute a Nonce command.
+     *
+     *  It loads data into Device Tempkey that is used as input in subsequent
+     *  Sign or Verify commands.
+     *
+     *  @param message  Data(generally message digest) to load in Tempkey.
+     *  @param len      message length.
+     *  @return         Error code from enum ATCAError.
+     */
+    ATCAError   Nonce(const uint8_t * message, size_t len);
+
+    /** Read 4 bytes from configuration zone.
+     *
+     *  @param address  Byte address in Data zone. Should be 4 byte aligned.
+     *  @param len      output buffer length.
+     *  @param obuf     output buffer of 4 bytes.
+     *  @return         Error code from enum ATCAError.
+     */
+    ATCAError   ReadConfig(uint8_t byte_address, size_t len, uint8_t * obuf);
+
+    /** Write 4 bytes or less in configuration zone. Writing across 4 byte word
```

Comment:
“Across” is the usual word here in my experience: “writing across a 4-byte word boundary is not allowed”. However, since the start address needs to be word-aligned, this is implied by the constraint that the length must be ≤4.